module/apmhttp/handler.go

```diff
@@ -116,11 +118,24 @@ func StartTransaction(tracer *apm.Tracer, name string, req *http.Request) (*apm.
 		traceContext.State, _ = ParseTracestateHeader(req.Header[TracestateHeader]...)
 	}
 	tx := tracer.StartTransactionOptions(name, "request", apm.TransactionOptions{TraceContext: traceContext})
+	tx.Context.SetHTTPRequest(req)
 	ctx := apm.ContextWithTransaction(req.Context(), tx)
 	req = RequestWithContext(ctx, req)
 	return tx, req
 }
 
+// StartTransactionWithBody returns a new Transaction with name,
+// created with tracer, and taking trace context from req.
+//
+// If the transaction is not ignored, the request and the request body
+// capturer will be returned with the transaction added to its context.
+func StartTransactionWithBody(tracer *apm.Tracer, name string, req *http.Request) (*apm.Transaction, *apm.BodyCapturer, *http.Request) {
+	tx, req := StartTransaction(tracer, name, req)
+	bc := tracer.CaptureHTTPRequestBody(req)
+	req = RequestWithContext(apm.ContextWithBodyCapturer(req.Context(), bc), req)
```

Comment:
```suggestion
	if bc != nil {
		req = RequestWithContext(apm.ContextWithBodyCapturer(req.Context(), bc), req)
	}
```

bc is nil if body capturing is disabled